src/librustc_errors/emitter.rs

```diff
@@ -1505,7 +1510,9 @@ impl EmitterWriter {
             let show_underline = !(parts.len() == 1 && parts[0].snippet.trim() == complete.trim())
                 && complete.lines().count() == 1;
 
-            let lines = sm.span_to_lines(parts[0].span).unwrap();
+            let lines = sm
+                .span_to_lines(parts[0].span)
```

Comment:
We should probably only skip one iteration in this loop if `span_to_lines` returns `Err` instead of introducing `is_valid_span`/`has_valid_spans`.
Other spans in the `suggestion` may be valid.

(Also need to avoid reporting anything if zero iterations is performed, unless it happens automatically.)